include/nanobind/nb_func.h

```diff
@@ -54,22 +60,61 @@ NB_INLINE PyObject *func_create(Func &&func, Return (*)(Args...),
         nargs = sizeof...(Args);
 
     // Determine the number of nb::arg/nb::arg_v annotations
-    constexpr size_t nargs_provided =
+    static constexpr size_t nargs_provided =
         ((std::is_same_v<arg, Extra> + std::is_same_v<arg_v, Extra>) + ... + 0);
-    constexpr bool is_method_det =
+    static constexpr bool is_method_det =
         (std::is_same_v<is_method, Extra> + ... + 0) != 0;
-    constexpr bool is_getter_det =
+    static constexpr bool is_getter_det =
         (std::is_same_v<is_getter, Extra> + ... + 0) != 0;
+    static constexpr bool has_arg_annotations =
+        nargs_provided > 0 && !is_getter_det;
+
+    static constexpr size_t
+        kwonly_pos_1 = index_1_v<std::is_same_v<kw_only, Extra>...>,
```

Comment:
What's the difference between `kwonly_pos_1` and `kwonly_pos_n`? Did you test the assertions below in the case of invalid parameter specifications?

Reply:
... just realized I accidentally was using index_1 in both. Fixed that and will test that multiple kw_only annotations produces an error.